src/compiler/transformers/es2015.ts

```diff
@@ -1571,6 +1578,78 @@ namespace ts {
                         break;
                 }
             }
+
+            /**
+             * Lazily creates the way class members will access the class prototype.
+             */
+            function getPrototypeStorageName() {
+                if (prototypeStorageName) {
+                    return prototypeStorageName;
+                }
+
+                const originalPrototypeAccess = createPropertyAccess(getInternalName(node), "prototype");
+
+                // If the class has exactly one non-static member, it'll access prototype members on itself:
+                //   ClassName.prototype.member = ...
+                if (!membersContainMultipleUniqueNames(node.members.filter(classMemberAssignsToPrototype))) {
+                    prototypeStorageName = originalPrototypeAccess;
+                    return prototypeStorageName;
+                }
+
+                // Since it has multiple non-static members, it'll store that prototype as a variable that can be minified:
+                //  var proto = ClassName.prototype;
+                //  proto.memberOne = ...
+                //  proto.memberTwo = ...
+                prototypeStorageName = createUniqueName("proto");
+
+                statements.splice(

```

Comment:
I'm not a fan of using `splice`, especially if the placement can be determined non-lazily.